pkg/sidecar/sidecar.go

```diff
@@ -183,6 +232,132 @@ func (s *Sidecar) dumpBundles(svidResponse *workloadapi.X509Context) error {
 	return nil
 }
 
+func (s *Sidecar) writeJSON(fileName string, certs map[string]interface{}) {
+	file, err := json.Marshal(certs)
+	if err != nil {
+		s.config.Log.Errorf("Unable to parse certs: %v", err)
+		return
+	}
+
+	jsonPath := path.Join(s.config.CertDir, fileName)
+	if err = os.WriteFile(jsonPath, file, os.ModePerm); err != nil {
+		s.config.Log.Errorf("Unable to write JSON file: %v", err)
+	}
+}
+
+func (s *Sidecar) updateJWTBundle(jwkSet *jwtbundle.Set) {
+	s.config.Log.Debug("Updating JWT bundle")
+
+	bundles := make(map[string]interface{})
+	for _, bundle := range jwkSet.Bundles() {
+		bytes, err := bundle.Marshal()
+		if err != nil {
+			s.config.Log.Errorf("Unable to marshal JWT bundle: %v", err)
+			continue
+		}
+		bundles[bundle.TrustDomain().Name()] = base64.StdEncoding.EncodeToString(bytes)
+	}
+
+	s.writeJSON(s.config.JWTBundleFilename, bundles)
+	s.config.Log.Info("JWT bundle updated")
```

Comment:
This log will be printed even if writeJSON fails. Suggest to make writeJSON return error and only print log if no error.

Reply:
I agree, just updated writeJSON, now only returns an error